### PR TITLE
ability to download results

### PR DESCRIPTION
**Pull request by apryka**
## Related tickets?

https://github.com/keen/keen-dataviz.js/issues/133

## How should this be tested?

Step through the code line by line. Things to keep in mind as you review:
 - Are there any edge cases not covered by this code?
 - Does this code follow conventions (naming, formatting, modularization, etc) where applicable?

Fetch the branch and/or deploy to staging to test the following:

- [ ] Does the code compile without warnings (check shell, console)?
- [ ] Do all tests pass?
- [ ] Does the UI, pixel by pixel, look exactly as expected (check various screen sizes, including mobile)?
- [ ] If the feature makes requests from the browser, inspect them in the Web Inspector. Do they look as expected (parameters, headers, etc)?
- [ ] If the feature sends data to Keen, is the data visible in the project if you run an extraction (include link to collection/query)?
- [ ] If the feature saves data to a database, can you confirm the data is indeed created in the database?

### CODE REVIEW

File: test/demo/keen-dataviz.css [hunk 1]
```diff
@@ -277,6 +277,21 @@
...
+    background-color: #f9f9f9;
+    border-color: #d6d6d6;
+    cursor: pointer;
+    -webkit-transition: background-color 300ms ease;
+    -o-transition: background-color 300ms ease;
+    transition: background-color 300ms ease
+  }
+.keen-dataviz-button:hover {
+  background-color: #E7E7E7;
+  background-color: #E7E7E7;
```
Comment (adamkasprowicz) [inline test/demo/keen-dataviz.css]: duplicated lines

File: test/demo/keen-dataviz.css [hunk 2]
```diff
@@ -699,11 +714,7 @@
     display: -ms-flexbox;
     display: flex;
     cursor: pointer;
-    overflow: hidden;
     padding: 0.2em;
-    white-space: nowrap;
-    -o-text-overflow: ellipsis;
-       text-overflow: ellipsis;
     font-size: 0.8em;
```
Comment (adamkasprowicz) [inline test/demo/keen-dataviz.css]: no color boxes in the legend anymore

File: lib/index.js [hunk 3]
```diff
@@ -558,6 +560,19 @@ Dataviz.prototype.prepare = function(){
...
+function renderDownloadButton({
+  element,
+  label = 'Download',
+  type = 'json',
+  data
+} = {}) {
+  const button = document.createElement('button');
+  button.innerText = label;
+  button.className = 'keen-dataviz-button';
+  button.onclick = event => downloadResults({ event, type, data});
```
Comment (adamkasprowicz) [inline lib/index.js]: let's use addEventListener, so in case somebody wants to attach any custom action to this button later - it'll be possible (eg. some people may want to attach google analytics tracking on this button)

File: lib/utils/download-results.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  const format = type.toLowerCase();
+  if (!supportedFormats.includes(type)) {
+    throw new Error('This type is not supported');
+  }
+
+  const fileName = 'chart';
+  let fileFormat = '';
+  let content = '';
+
+  if (format === 'png') {
```
Comment (adamkasprowicz) [inline lib/utils/download-results.js]: json and csv download works, but png/jpg doesnt 

FileSaver.min.js:1 Uncaught (in promise) TypeError: Failed to execute 'createObjectURL' on 'URL': No function was found that matched the signature provided.
    at a (FileSaver.min.js:1)
    at export-svg.js:10
